ecosystem/sf-indexer/aptos-substreams/src/lib.rs

```diff
@@ -85,26 +88,71 @@ fn block_to_block_output(input_block: Block) -> Result<BlockOutput, Error> {
     })
 }
 
-#[substreams::handlers::store]
-fn store_count(transaction: Transaction, store: store::StoreAddInt64) {
-    store.add(transaction.version, generate_trx_key(), 1);
-    store.add(
-        transaction.version,
-        generate_trx_type_key(transaction.r#type()),
-        1,
-    );
-}
-
-fn generate_trx_key() -> String {
-    String::from("total")
-}
-
-fn generate_trx_type_key(trx_type: TransactionType) -> String {
-    match trx_type {
-        TransactionType::Genesis => "genesis",
-        TransactionType::BlockMetadata => "block_metadata",
-        TransactionType::StateCheckpoint => "state_checkpoint",
-        TransactionType::User => "user",
+#[substreams::handlers::map]
+fn block_output_to_token(block: BlockOutput) -> Result<Tokens, Error> {
+    let mut tokens = vec![];
+    // let mut token_ownerships = vec![];
+    let mut token_datas = vec![];
+    let mut collection_datas = vec![];
+    for txn in block.transactions {
+        let txn_version = txn.transaction_info_output.unwrap().version;
+        // Claims will be supported separately
+        let supported_functions = vec![
```

Comment:
would we be able to index data from other modules calling our public functions? for example, we have a public function here https://github.com/aptos-labs/aptos-core/blob/7131fad47d92b893dadaa2125e4cae73f82e680d/aptos-move/framework/aptos-token/sources/token.move#L483. Other contracts can call to transfer token. In this case, the function is not the entry function.

Reply:
Yup, I changed the logic. Basically if we somehow can't find the owner address we'll just index the rest of the token and leave owner address as missing. Rn it'd be missing for claim / offer for example. This way we can be a lot more generic.

Reply:
all the data from public function being called by other modules are filtered out right? there are other public function that could be called ex: this is to create more tokens https://github.com/aptos-labs/aptos-core/blob/8f8de4adcdaa0edbac2cb3c809c278d97331bdba/aptos-move/framework/aptos-token/sources/token.move#L751

Reply:
hmm idk what your question is sorry. if the token is 0x3 i'm assuming that we should index it. is this not the right assumptino?

Reply:
I see the filter is deleted, which make senses.